src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -562,12 +562,15 @@ private static bool CanBindToTheseConstructorParameters(ParameterInfo[] construc
         // in their config class, then it is cloned to a new dictionary, the same way as other collections.
         [RequiresDynamicCode(DynamicCodeWarningMessage)]
         [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the value objects in the dictionary so its members may be trimmed.")]
-        private static void BindConcreteDictionary(
+        private static void BindDictionary(
             object? dictionary,
```

Comment:
(nit, existing)

```suggestion
            object dictionary,
```

This method is only ever called with a non-null object, and we have an assert that it is not null. Seems like it makes more sense to codify this in the signature.